.azure-pipelines/pipelines.yml

```diff
@@ -103,7 +103,28 @@ jobs:
           bazelBuildExtraOptions: "--test_env=ENVOY_IP_TEST_VERSIONS=v4only --curses=no"
 
       - script: ci/run_envoy_docker.sh 'ci/upload_gcs_artifact.sh /build/envoy/generated/coverage coverage'
-        displayName: "Upload Report to GCS"
+        displayName: "Upload Coverage Report to GCS"
+        env:
+          ENVOY_DOCKER_BUILD_DIR: $(Build.StagingDirectory)
+          GCP_SERVICE_ACCOUNT_KEY: $(GcpServiceAccountKey)
+          GCS_ARTIFACT_BUCKET: $(GcsArtifactBucket)
+        condition: always()
+
+  - job: fuzz_coverage
```

Comment:
I think you can make this [coverage, fuzz_coverage] to matirx? (see bazel job above)